src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Tag.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Binary;
+using System.Globalization;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal partial class CborReader
+    {
+        public CborTag ReadTag()
+        {
+            CborTag tag = PeekTagCore(out int additionalBytes);
+
+            AdvanceBuffer(1 + additionalBytes);
+            _isTagContext = true;
+            return tag;
+        }
+
+        public void ReadTag(CborTag expectedTag)
+        {
+            CborTag tag = PeekTagCore(out int additionalBytes);
+
+            if (expectedTag != tag)
+            {
+                throw new InvalidOperationException("CBOR tag mismatch.");
+            }
+
+            AdvanceBuffer(1 + additionalBytes);
+            _isTagContext = true;
+        }
+
+        public CborTag PeekTag() => PeekTagCore(out int _);
+
+        private CborTag PeekTagCore(out int additionalBytes)
+        {
+            CborInitialByte header = PeekInitialByte(expectedType: CborMajorType.Tag);
+            return (CborTag)ReadUnsignedInteger(_buffer.Span, header, out additionalBytes);
+        }
+
+        // Additional tagged type support
+
+        public DateTimeOffset ReadDateTimeOffset()
+        {
+            // implements https://tools.ietf.org/html/rfc7049#section-2.4.1
+
+            switch (ReadTag())
+            {
+                case CborTag.DateTimeString:
+                    if (Peek() != CborReaderState.TextString)
+                    {
+                        throw new FormatException("String DateTime semantic tag should annotate string value.");
+                    }
+
+                    string dateString = ReadTextString();
+                    return DateTimeOffset.Parse(dateString, formatProvider: null, DateTimeStyles.RoundtripKind);
```

Comment:
There's no baked-in format string for exclusively recognizing RFC3339 dates. As such, I'm using `DateTimeOffset.Parse` instead of `DateTimeOffset.ParseExact`, at the cost of tolerating certain nonstandard string formats. I'm assuming this shouldn't be a big problem given this is a reader.

Reply:
Isn't it just `"yyyy-MM-ddTHH:mm:ss.sssK"`?

Reply:
Maybe, do you think it would be worth enforcing on the reader's side?

Reply:
Will revisit this issue when implementing conformance levels.